### PR TITLE
xtensa/esp32: Fix some Wi-Fi issues

### PR DESCRIPTION
**Pull request by cwespressif**
## Summary

    1. Fix the issue that Wi-Fi can't connect to some special routers occasionally.
    2. Update Wi-Fi driver code to fix issue of failure to send pkt.
    3. Replace software random with hardware random
  
## Impact

Fix some ESP32 Wi-Fi driver issues

## Testing

`esp32-devkitc`

### CODE REVIEW

File: boards/xtensa/esp32/esp32-devkitc/src/esp32_reset.c [hunk 1]
```diff
@@ -24,11 +24,20 @@
...
+#include <assert.h>
 #include <nuttx/arch.h>
 #include <nuttx/board.h>
 
+#include "esp32_systemreset.h"
+
 #ifdef CONFIG_BOARDCTL_RESET
 
+#if CONFIG_BOARD_ASSERT_RESET_VALUE == EXIT_SUCCESS
+#  error "CONFIG_BOARD_ASSERT_RESET_VALUE is equal to EXIT_SUCCESS"
```
Comment (gustavonihei) [inline boards/xtensa/esp32/esp32-devkitc/src/esp32_reset.c:38]:
```suggestion
#  error "CONFIG_BOARD_ASSERT_RESET_VALUE must not be equal to EXIT_SUCCESS"
```
Suggestion for making the error message clearly state the reason for failure.

File: arch/xtensa/src/esp32/esp32_systemreset.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+ * Input Parameters:
+ *   handler - Function to execute on restart
+ *
+ * Returned Value:
+ *   OK on success (positive non-zero values are cmd-specific)
+ *   Negated errno returned on failure.
+ *
+ ****************************************************************************/
+
+int up_register_shutdown_handler(shutdown_handler_t handler);
```
Comment (Ouss4) [inline arch/xtensa/src/esp32/esp32_systemreset.h:69]:
```suggestion
int esp32_register_shutdown_handler(shutdown_handler_t handler);
```

Let's please use the `esp32_` prefix as these functions are not exported outside the arch directory.

Comment (cwespressif) [reply arch/xtensa/src/esp32/esp32_systemreset.h]: Done, thanks.

File: boards/xtensa/esp32/esp32-devkitc/src/esp32_reset.c (hunk 1 above)
Comment (cwespressif) [reply boards/xtensa/esp32/esp32-devkitc/src/esp32_reset.c]: Done, thanks

**Comment by acassis** [comment]
I noticed we CI is failing often with this TLS timeout:
```
Error response from daemon: Head "https://ghcr.io/v2/apache/incubator-nuttx/apache-nuttx-ci-linux/manifests/latest": Get "https://ghcr.io/token?account=%24%7BGITHUB_ACTOR%7D&scope=repository%3Aapache%2Fincubator-nuttx%2Fapache-nuttx-ci-linux%3Apull&service=ghcr.io": net/http: TLS handshake timeout
```

Any idea how to fix it?

**Comment by xiaoxiang781216** [comment]
rebase to mainline and force push again.

**Comment by cwespressif** [comment]
> rebase to mainline and force push again.

Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**